File: Version-1/C++/Question.cpp
#include "Question.h"
#include <iostream>

// static variable definitions
int Question::nQuestion = 0;
int Question::nCorrect = 0;

void Question::check() {
    nQuestion++;
    std::string answer = ask();

    if (answer == correctAnswer) {
        std::cout << "Correct Answer!\n";
        nCorrect++;
    } else {
        std::cout << "Incorrect. The correct answer is A\n";
    }
}

void Question::showResults() {
    std::count << nCorrect << " Correct out of "
	           << nQuestion << " question\n";
}

File: Version-1/Java/Question.java
import javax.swing.JOptionPane;

public abstract class Question {
    static int nQuestion = 0;
    static int nCorrect = 0;

    abstract String ask();

    String question;
    String correctAnswer;

    void check() {
        nQuestion++;
	String answer = ask(); // Will use the method defined in the subclasses
        if (answer.equals(correctAnswer)) {
	    JOptionPane.showMessageDialog(null, "Correct Answer! ");
	    nCorrect++;
        } else {
	    JOptionPane.showMessageDialog(null, "Incorrect. The correct answer is " + "A");
	}
    }

    static void showResults() {
        JOptionPane.showMessageDialog(null, nCorrect + " Correct out of " + nQuestion + " questions");
    }
}
